src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/DeliverySessionController.kt

```diff
@@ -118,4 +133,88 @@ class DeliverySessionController(
     val user = userMapper.fromToken(authentication)
     return DeliverySessionDTO.from(deliverySessionService.submitSessionFeedback(actionPlanId, sessionNumber, user))
   }
+
+  @GetMapping("/referral/{referralId}/delivery-session-appointments/{appointmentId}")
+  fun getDeliverySessionAppointment(
+    @PathVariable(name = "referralId") referralId: UUID,
+    @PathVariable(name = "appointmentId") appointmentId: UUID,
+    authentication: JwtAuthenticationToken,
+  ): DeliverySessionAppointmentDTO {
+    val user = userMapper.fromToken(authentication)
+    val referral = referralService.getSentReferral(referralId) ?: throw ResponseStatusException(
+      HttpStatus.BAD_REQUEST, "sent referral not found [referralId=$referralId]"
+    )
+    referralAccessChecker.forUser(referral, user)
+    val matchingAppointment = deliverySessionService.getSessions(referralId)
+      .flatMap { session -> session.appointments.map { appointment -> Pair(session.sessionNumber, appointment) } }
+      .firstOrNull { appointment -> appointment.second.id == appointmentId }
```

Comment:
@andrewburrowsmoj @mahalrmoj In non-Boot environments, I've seen this pattern dealt by using custom queries in the service, which feels like a more efficient approach.

I don't quite understand the need for this endpoint though -- `/referral/{id}/sessions/{sessionNumber}` seems ok and doesn't use action plans. Is this endpoint necessary?